### PR TITLE
SSR support

### PR DESCRIPTION
**Pull request by fsoikin**
* Ported SSR support from the CV App
* Added comments
* Cleaned up some naming
* Simplified the `JsCallback` API

### CODE REVIEW

File: src/Elmish/Boot.purs [hunk 1]
```diff
@@ -1,61 +1,125 @@
...
+-- |        contain pre-rendered HTML inside. See React docs for more
+-- |        on server-side rendering: https://reactjs.org/docs/react-dom.html#hydrate
+-- |     * `renderToString` -
+-- |        meant to be called on the server (e.g. by running the code
+-- |        under NodeJS) to perform the server-side render. Takes props¹
+-- |        and returns a `String` containing the resulting HTML.
+-- |
+-- | The idea is that the PureScript code would export such `BootRecord` for
+-- | consumption by bootstrap JavaScript code in the page and/or server-side
+-- | NodeJS code (which could be written in PureScript or not). For "plan React"
```
Comment (mcordova47) [inline src/Elmish/Boot.purs:22]:
```suggestion
-- | NodeJS code (which could be written in PureScript or not). For "plain React"
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,31 @@ All notable changes to this project will be documented in this file.
...
+  `boot'`. Replaced and subsumed by server-side rendering support (see "Added").
+
+## Changed
+
+- **Breaking**: `Elmish.React.reactMount` renamed to `render` to match React's
+  naming.
+- `wrapWithLocalState` will now report errors to the console instead of
+  swallowing them. This could be used by catch-all error reporters such as
+  Rollbar or Airbrake.
+- **Breaking**: `mkJsCallback` remaned to `jsCallback` and lost its `onError`
```
Comment (sassela) [inline CHANGELOG.md:27]: `renamed`

File: src/Elmish/JsCallback.purs [hunk 3]
```diff
@@ -63,53 +60,61 @@ jsCallback0 fn = mkJsCallback fn ignoreErrors
...
 -- |      // JSX:
 -- |      export const TheView = props =>
 -- |        <div>
 -- |          <button onClick={props.onSave}>Save</button>
 -- |          <button onClick={props.onCancel}>Cancel</button>
--- |          <button onClick={() => props.onFoo("bar", "baz")}>Foo</button>
+-- |          <button onClick={() => props.onFoo("bar", 42)}>Foo</button>
 -- |        </div>
 -- |
 -- | In this example, the parameters `bar` and `baz` will undergo validation at
```
Comment (sassela) [inline src/Elmish/JsCallback.purs:77]: `baz` was removed from the example, but still referred to here.

File: CHANGELOG.md (hunk 2 above)
Comment (fsoikin) [reply CHANGELOG.md:27]: Oops

File: src/Elmish/JsCallback.purs (hunk 3 above)
Comment (fsoikin) [reply src/Elmish/JsCallback.purs:77]: `baz` hasn't been removed, it was just changed from a string to an int, which is how it's expected to be on the PureScript side: https://github.com/collegevine/purescript-elmish/pull/19/files#diff-d900e91a5a6c6d7baa84f2fc05fe8a8bR65-R66

The original example had a mistake in it.

**Pull request merged**